walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,129 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Arrays;
 
+/**
+* The FindMeetingQuery class represents a meeting that we want to schedule given
+* the TimeRanges of events. 
+*/
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+  
+  /**
+  * Returns an unsorted ArrayList of TimeRanges that represent the times a group of attendees
+  * can meet.
+  */
+  private ArrayList<TimeRange> getAvailability(ArrayList<TimeRange> busyTimesForChosenAttendees) {
+    TimeRange currentBusyTime = busyTimesForChosenAttendees.get(0);
+    ArrayList<TimeRange> busyTimesForChosenAttendeesNoOverlaps = new ArrayList<>();
+    // Stores all TimeRanges in which attendees can meet.
+    ArrayList<TimeRange> freeTimesForChosenAttendees = new ArrayList<>();
+    
+    for (TimeRange time : busyTimesForChosenAttendees) {
+      // Checks whether our attendees have overlapping TimeRanges.
+      if (time.overlaps(currentBusyTime)) {
+        int start = currentBusyTime.start();
+        currentBusyTime = TimeRange.fromStartEnd(start, time.end() < currentBusyTime.end() ? currentBusyTime.end() : time.end(), false);
```

Comment:
I just realized that instead of `time.end() < currentBusyTime.end() ? currentBusyTime.end() : time.end()` you could use a max function here (I'm guessing Java has one built in). Honestly fine either way though.